src/view/dataElements/xdmObject/components/sandboxSelector.jsx

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2021 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+import React, { useContext, useEffect } from "react";
+import { useAsyncList } from "@react-stately/data";
+import { Item, Picker } from "@adobe/react-spectrum";
+import PropTypes from "prop-types";
+import fetchSandboxes from "../helpers/fetchSandboxes";
+import FieldDescriptionAndError from "../../../components/fieldDescriptionAndError";
+import useReportAsyncError from "../../../utils/useReportAsyncError";
+import useIsFirstRender from "../../../utils/useIsFirstRender";
+import ExtensionViewContext from "../../../components/extensionViewContext";
+
+const getKey = sandbox => sandbox && sandbox.name;
+const getLabel = sandbox => {
+  if (!sandbox) {
+    return undefined;
+  }
+  const region = sandbox.region ? ` (${sandbox.region.toUpperCase()})` : "";
+  return `${sandbox.type.toUpperCase()}  ${sandbox.title}${region}`;
+};
+
+const SandboxSelector = ({
+  defaultSelectedSandbox,
+  onSelectionChange,
+  errorMessage
+}) => {
+  const {
+    initInfo: {
+      company: { orgId },
+      tokens: { imsAccess }
+    }
+  } = useContext(ExtensionViewContext);
+  const reportAsyncError = useReportAsyncError();
+  const isFirstRender = useIsFirstRender();
+
+  const sandboxList = useAsyncList({
+    async load({ signal }) {
+      let sandboxes;
+      try {
+        ({ results: sandboxes } = await fetchSandboxes({
+          orgId,
+          imsAccess,
+          signal
+        }));
+      } catch (e) {
+        if (e.name !== "AbortError") {
+          console.error(e);
+          reportAsyncError(new Error(`Failed to load sandboxes. ${e.message}`));
```

Comment:
I did consider moving console.error into `reportAsyncError` and having `reportAsyncError` also accept something like `errorMessage` (which would be `Failed to load sandboxes` in this case) and `originatingError` (which would be `e` in this case), but it felt like we were overloading `reportAsyncError`. We could make an abstraction on top of `reportAsyncError`, but I was a bit ambivalent. You'll also notice in other parts of the code that we don't always have an "originating error".